### PR TITLE
Unit tests for Pen class

### PR DESCRIPTION
**Pull request by ancheypeng**
This pull request adds unit tests for some of the functions in the Pen class. 

The following unit tests have been implemented:

- test_reset_manure
- test_calc_avg_growth
- test_clear
- test_subset_class_feeds

### CODE REVIEW

**Comment by KFosterReed** [comment]
This runs and reading through the tests it makes logical sense to me but I am going to defer to @PooyaHekmati. After he has approved them, if any further changes were made, I will run the model again and then give my approval. Great work Anchey!

File: tests/animal_config.json
Comment (PooyaHekmati) [inline tests/animal_config.json]: Just a thought: can we / does it make sense to put these numbers in the same line and reduce the number of lines in the file?

File: tests/expected_outputs/expected_feed_units.py
Comment (PooyaHekmati) [inline tests/expected_outputs/expected_feed_units.py]: I think these can go in the same line.

File: tests/test_pen.py
Comment (PooyaHekmati) [inline tests/test_pen.py]: Missed pass and type annotation here

File: tests/test_pen.py
Comment (PooyaHekmati) [inline tests/test_pen.py]: Can we use zip or something similar to avoid duplicated code here?

**Review by PooyaHekmati — CHANGES_REQUESTED** [review]
Overall, looks good; good job! In addition to the minor comments I made, I do not see expected_feed_units.py being used anywhere. What's the plan for it?

**Comment by ancheypeng** [comment]
> Overall, looks good; good job! In addition to the minor comments I made, I do not see expected_feed_units.py being used anywhere. What's the plan for it?

The expected_feed_units.py and the animal_config.json were files used for old code/other tests that have been removed/replaced, but I forgot to remove them.

File: tests/test_pen.py
Comment (ancheypeng) [reply tests/test_pen.py]: This unit test was not implemented yet, so I just removed the pen parameter.

File: tests/test_pen.py
Comment (ancheypeng) [reply tests/test_pen.py]: I couldn't think of a good way to reduce the duplication just in that section because each set has slightly different contents from all the others. Instead, I pushed a new change which changes the parameterization overall. However, I'm not sure if it's the best practice because I had to use a "fake" fixture (feed_allocations) because a real fixture did not work well with the pytest parameterization.

**Review by PooyaHekmati — APPROVED** [review]

**Review by KFosterReed — APPROVED** [review]
All the tests make logical sense to me and I was able to break a couple of them by changing what I thought it was testing in the function. Well done!

**Pull request merged**
